evm/src/cross_table_lookup.rs

```diff
@@ -21,33 +22,61 @@ use crate::proof::{StarkProofWithPublicInputs, StarkProofWithPublicInputsTarget}
 use crate::stark::Stark;
 use crate::vars::{StarkEvaluationTargets, StarkEvaluationVars};
 
+#[derive(Clone)]
+pub struct TableWithColumns {
+    table: Table,
+    columns: Vec<usize>,
+    /// Vector of columns `[c_1,...,c_k]` used as a filter using the sum `c_1 + ... + c_k`.
+    /// An empty vector corresponds to no filter.
+    filter_columns: Vec<usize>,
+}
+
+impl TableWithColumns {
+    pub fn new(table: Table, columns: Vec<usize>, filter_columns: Vec<usize>) -> Self {
+        debug_assert_eq!(
+            filter_columns.iter().collect::<HashSet<_>>().len(),
+            filter_columns.len(),
+            "Duplicate filter columns."
+        );
+        Self {
+            table,
+            columns,
+            filter_columns,
+        }
+    }
+}
+
 #[derive(Clone)]
 pub struct CrossTableLookup<F: Field> {
-    looking_tables: Vec<Table>,
-    looking_columns: Vec<Vec<usize>>,
-    looked_table: Table,
-    looked_columns: Vec<usize>,
-    default: Vec<F>,
+    looking_tables: Vec<TableWithColumns>,
+    looked_table: TableWithColumns,
+    /// Default value if filters are not used.
+    default: Option<Vec<F>>,
 }
 
 impl<F: Field> CrossTableLookup<F> {
     pub fn new(
-        looking_tables: Vec<Table>,
-        looking_columns: Vec<Vec<usize>>,
-        looked_table: Table,
-        looked_columns: Vec<usize>,
-        default: Vec<F>,
+        looking_tables: Vec<TableWithColumns>,
+        looked_table: TableWithColumns,
+        default: Option<Vec<F>>,
     ) -> Self {
-        assert_eq!(looking_tables.len(), looking_columns.len());
-        assert!(looking_columns
+        assert!(looking_tables
             .iter()
-            .all(|cols| cols.len() == looked_columns.len()));
-        assert!(default.len() == looked_columns.len());
+            .all(|twc| twc.columns.len() == looked_table.columns.len()));
+        assert!(
+            looking_tables
+                .iter()
+                .all(|twc| twc.filter_columns.is_empty())
```

Comment:
nit: might be best to move the `== default.is_some()` inside `all`, so that we do the check for each looking table.